packages/common/src/base64.js

```diff
@@ -20,10 +19,7 @@ export function fromBase64url(str) {
  * @returns {string}
  */
 export function toBase64Url(str) {
-  return btoa(str)
-    .replace(/\+/g, `-`)
-    .replace(/\//g, `_`)
-    .replace(/\=/g, ``);
+  return UrlSafeBase64.encode(new Buffer(str));
```

Comment:
if str is already a buffer, this could effectively be `new Buffer(Buffer(string))`. Can you add a check to make sure `new Buffer()` is only called if str is not a buffer?